dra-evolution/pkg/api/capacity_types.go

```diff
@@ -3,168 +3,110 @@ package api
 import (
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 )
 
-const (
-	DevMgmtAPIVersion = "devmgmtproto.k8s.io/v1alpha1"
-)
-
-// DevicePool represents a collection of devices managed by a given driver. How
+// ResourcePool represents a collection of devices managed by a given driver. How
 // devices are divided into pools is driver-specific, but typically the
 // expectation would a be a pool per identical collection of devices, per node.
 // It is fine to have more than one pool for a given node, for the same driver.
-type DevicePool struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   DevicePoolSpec   `json:"spec,omitempty"`
-	Status DevicePoolStatus `json:"status,omitempty"`
-}
-
-// DevicePoolSpec identifies the driver and contains the data for the pool
-// prior to any allocations.
-// NOTE: It's not clear that spec/status is the right model for this data.
-type DevicePoolSpec struct {
-	// NodeName is the name of the node containing the devices in the pool.
-	// For network attached devices, this may be empty.
+//
+// Where a device gets published may change over time. The unique identifier
+// for a device is the tuple `<driver name>/<node name>/<device name>`. Each
+// of these names is a DNS label or domain, so it is okay to concatenate them
+// like this in a string with a slash as separator.
+type ResourcePool struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
 	// +optional
-	NodeName *string `json:"nodeName,omitempty"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// Driver is the name of the DeviceDriver that created this object and
-	// owns the data in it.
-	// +required
-	Driver string `json:"driver,omitempty"`
+	Spec ResourcePoolSpec `json:"spec"`
```

Comment:
What we have in status is the health of the device. This would be a condition, typically, and could be sparsely populated (no need to populate it for healthy devices). Initially, this would be the driver reporting both the details of the device, and its status. However, we do have use cases for reporting a device as "degraded" in its performance. That may be done by a monitoring agent, the workload itself, a controller processing metrics data or even a human. That's a future consideration though, not in the immediate plans.

Reply:
Sorry my comment was not in response to Tim but rather the original. The timestamp GH puts on the comment is based on when originally written, not when the review is submitted (which seems bad to me...).

Anyway, my understanding of our discussion today was that we would have both spec and status.

Reply:
I think we agreed to keep spec because status is very likely, one way or another.

Regarding Tim's comment: "not usable" would indeed go into the status because that might be a decision made elsewhere. The clearest use case is perhaps an admin who temporarily takes a device offline.

Autoscalers also will be affected. For "the" Cluster Autoscaler it won't matter, it will just reuse the in-tree scheduler code. Others will have to extend their logic whenever we make such changes (also applies to any other API change).